clients/docs/java.rst

```diff
@@ -0,0 +1,317 @@
+
+#. View the :devx-examples:`producer Avro code|clients/cloud/java/src/main/java/io/confluent/examples/clients/cloud/ProducerAvroExample.java`.
+
+
+Consume Avro Records
+~~~~~~~~~~~~~~~~~~~~
+
+#. Run the Avro consumer, passing in arguments for:
+
+   - the local file with configuration parameters to connect to your Kafka cluster
+   - the topic name
+
+   .. code-block:: text
+
+      # Compile the Java code
+      mvn clean package
+
+      # Run the Avro consumer
+      mvn exec:java -Dexec.mainClass="io.confluent.examples.clients.cloud.ConsumerAvroExample" \
+      -Dexec.args="$HOME/.confluent/java.config test2"
+
+#. View the :devx-examples:`consumer Avro code|clients/cloud/java/src/main/java/io/confluent/examples/clients/cloud/ConsumerAvroExample.java`.
+
+
+Avro Kafka Streams
+~~~~~~~~~~~~~~~~~~
+
+#. Run the Avro Kafka Streams application, passing in arguments for:
+
+   -  the local file with configuration parameters to connect to your Kafka
+      cluster
+   -  the same topic name you used earlier
+
+   .. code-block:: text
+
+      # Compile the Java code
+      mvn clean package
+
+      # Run the Avro Kafka streams application
+      mvn exec:java -Dexec.mainClass="io.confluent.examples.clients.cloud.StreamsAvroExample" \
+      -Dexec.args="$HOME/.confluent/java.config test2"
+
+#. View the :devx-examples:`Kafka Streams Avro code|clients/cloud/java/src/main/java/io/confluent/examples/clients/cloud/StreamsAvroExample.java`.
+
+
+Schema Evolution with Confluent Cloud Schema Registry
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+#. View the schema information registered in |ccloud| |sr|. In the output below,
+   substitute values for  ``{{ SR_API_KEY }}``, ``{{ SR_API_SECRET }}``, and
+   ``{{ SR_ENDPOINT }}``.
+
+   .. code-block:: text
+
+      # View the list of registered subjects
+      curl -u {{ SR_API_KEY }}:{{ SR_API_SECRET }} https://{{ SR_ENDPOINT }}/subjects
+      ["test2-value"]
+
+      # View the schema information for subject `test2-value`
+      curl -u {{ SR_API_KEY }}:{{ SR_API_SECRET }} https://{{ SR_ENDPOINT }}/subjects/test2-value/versions/1
+      {"subject":"test2-value","version":1,"id":100001,"schema":"{\"name\":\"io.confluent.examples.clients.cloud.DataRecordAvro\",\"type\":\"record\",\"fields\":[{\"name\":\"count\",\"type\":\"long\"}]}"}
+
+#. For schema evolution, you can `test schema
+   compatibility
+   <https://docs.confluent.io/current/schema-registry/develop/maven-plugin.html?utm_source=github&utm_medium=demo&utm_campaign=ch.examples_type.community_content.clients-ccloud#schema-registry-test-compatibility>`__
+   between newer schema versions and older schema versions in |ccloud| |sr|. The
+   `pom.xml <pom.xml>`__ hardcodes the Schema Registry subject name to
+   ``test2-value``—change this if you didn't use topic name ``test2``. Then
+   test local schema compatibility for
+   `src/main/resources/avro/io/confluent/examples/clients/cloud/DataRecordAvro2a.avsc
```

Comment:
Next 6 lines should be replaced to the following

```
:devx-example:`DataRecordAvro2a.avsc|clients/cloud/java/src/main/resources/avro/io/confluent/examples/clients/cloud/DataRecordAvro2a.avsc`, which should fail, and
:devx-example:`DataRecordAvro2b.avsc|clients/cloud/java/src/main/resources/avro/io/confluent/examples/clients/cloud/DataRecordAvro2b.avsc`, which should pass.
```

Reply:
Updated